design/specs/foreign-key-constraints.md

```diff
@@ -2,75 +2,169 @@
+- Once questions are answered,  a summary of the system's changes is displayed in a section titled 'Under the Hood.'
+- Before creating the link, the user will have the chance to rename the new columns or tables. Invalid column or table names should display an inline warning and prevent saving.
 
-**Note: This step does not cover the scenario where a user has a unique table, in which case they need to create an additional table to add a relationship.**
+### Scenario 1b: Manually from the 'Table Constraints' settings
 
-#### 2. The user selects a table from the list of available tables
-- The user selects the table on the 'one side of the 'one-to-many' relationship.
+#### Steps for 1b
 
-#### 3. The user selects a column belonging to the selected table to create the link
-- The primary key column is pre-selected by default
-#### The user sets the new column as a foreign key for a column in the 'artist' table
-- If the column had values, this action would clear those unless they exactly matched contents from the linked column
-#### The user retrieves records from the linked table and adds them to the new column
-- Autocomplete should allow users to complete this action without additional clicks
-- When no results are available, we should show an option to add a new record
-#### 4. The user removes the foreign key constraint from the column
-- The system will assign the column the type of the column that was linked (if the linked column was a duration type, then it should be assigned as such)
+- A user wants to set up a foreign key constraint to single or multiple columns to a currently active table.
+- The user opens the table options by clicking on the table name label in the toolbar area. From the menu, the user selects the option 'Table Constraints'
+- This opens a dialog from which the user can set all supported constraint types.
+- The user sees a layout with two panels. On one side, all the existing constraints are listed. On the other side, a form with the constraints configuration is displayed. The form corresponds to the currently active list item.
+- The list panel contains actions at the bottom of the panel. From there, the user can choose to add or delete an item from the list. The user clicks on 'Add' to create a new constraint.
+- The user is presented with a form. From there, they select the type of constraint to be added. The user selects the 'Foreign Key' constraint option from the list.
+- The user selects single or multiple columns they wish to apply the constraint to.
+- The user selects a table to be referenced by these columns.
+- The user selects a column in the reference table from which the column will match values. By default, this is set to the 'Primary Key' column in the referenced table. When changing this field, the user can select only columns with a UNIQUE or PRIMARY KEY constraint.
 
----
+## Scenario 2: User Edits an Existing Foreign Key Constraint
```

Comment:
At the DB level, altering foreign keys isn't possible (at least for the things we're actually defining for foreign key constraints through the UI).  Is it really needed?  The DB process to do this would be to create a new foreign key constraint, and drop the old.  This would need to involve some transactions that would make doing it step-by-step ill-advised, since it would involve repetitively creating and dropping foreign key constraints.  These operations would each lock multiple tables for reading and writing.

Thus, I think we should consider focusing on helping users go through the "drop constraint", then "create new constraint" process smoothly instead.  This will avoid confusion, in particular around potential errors arising from a different process occurring under the hood than is portrayed by the UI.

Reply:
Agree with @mathemancer. The foreign key constraint should not be updatable and should rather point to deleting the constraint and creating a different one. Instead of apply button something like delete and copy template(in case if all they want is to include a constraint on an additional column from the same table)

Reply:
All constraints.